### PR TITLE
add comparison function for Pp.Ast.t

### PR DESCRIPTION
**Pull request by Alizter**

### CODE REVIEW

File: src/pp.ml [hunk 1]
```diff
@@ -22,6 +22,70 @@ module Ast = struct
...
+    let r = f a c in
+    if r <> 0 then
+      r
+    else
+      g b d
+
+  let compare_triple f g h (x, y, z) (a, b, c) =
+    compare_both f (compare_both g h) (x, (y, z)) (a, (b, c))
+
+  let compare compare_tag =
```
Comment (rgrinberg) [inline src/pp.ml]: You can define it directly on `Pp.t`

Comment (Alizter) [reply src/pp.ml]: I can't compare `Format` so we can just use this together with `to_ast` for Pp.

Comment (rgrinberg) [reply src/pp.ml]: No. that's unnecessarily slow. Just raise if you get `Format`. We should just get rid of that constructor

Comment (Alizter) [reply src/pp.ml]: I'll remove `Format` in another PR. I've made `Pp.compare` which will raise `Invalid_argument` if two `Pp.of_fmt` values are provided.

Comment (rgrinberg) [reply src/pp.ml]: To clarify, I made a mistake not listening to Jeremie here: https://github.com/ocaml-dune/pp/pull/1

What I think we can do is introduce a `Pp.Fmt` submodule that allows to inject this `Format` constructor. The core `Pp.t` should stay serializable, comparable, and all that good stuff.

Comment (Alizter) [reply src/pp.ml]: That seems sensible, but it is beyond the scope of what I am willing to do at the moment. We should definitely take another look at that in the future.

Comment (rgrinberg) [reply src/pp.ml]: Ok, could you add a CHANGES entry?

**Comment by Alizter** [comment]
@rgrinberg Added a CHANGES entry. (Before I also added a test).

**Pull request merged**